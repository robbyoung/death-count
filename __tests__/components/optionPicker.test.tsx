import React from 'react';
import renderer from 'react-test-renderer';
import OptionPicker from '../../app/components/optionPicker';

describe('Option Picker Component', () => {
    it('can render an option picker with nothing selected', () => {
        const component = renderer.create(<OptionPicker
            onSelect={() => undefined}
            options={[{
                key: 'key 1',
                value: 'value 1',
            }, {
                key: 'key 2',
                value: 'value 2',
            }]}
            selected={undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('can render an option picker with a selected item', () => {
        const component = renderer.create(<OptionPicker
            onSelect={() => undefined}
            options={[{
                key: 'key 1',
                value: 'value 1',
            }, {
                key: 'key 2',
                value: 'value 2',
            }]}
            selected={'value 1'}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});