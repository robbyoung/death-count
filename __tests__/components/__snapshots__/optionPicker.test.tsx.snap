// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Picker Component can render an option picker with a selected item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "margin": 10,
    }
  }
>
  <View>
    <RCTPicker
      items={
        Array [
          Object {
            "label": "key 1",
            "textColor": undefined,
            "value": "value 1",
          },
          Object {
            "label": "key 2",
            "textColor": undefined,
            "value": "value 2",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Option Picker Component can render an option picker with nothing selected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "margin": 10,
    }
  }
>
  <View>
    <RCTPicker
      items={
        Array [
          Object {
            "label": "None Selected",
            "textColor": undefined,
            "value": undefined,
          },
          Object {
            "label": "key 1",
            "textColor": undefined,
            "value": "value 1",
          },
          Object {
            "label": "key 2",
            "textColor": undefined,
            "value": "value 2",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
